op-batcher/batcher/channel.go

```diff
@@ -85,9 +85,9 @@ func (c *channel) TxConfirmed(id string, inclusionBlock eth.BlockID) bool {
 		c.log.Info("Channel is fully submitted", "id", c.ID(), "min_inclusion_block", c.minInclusionBlock, "max_inclusion_block", c.maxInclusionBlock)
 	}
 
-	// If this channel timed out, put the pending blocks back into the local saved blocks
-	// and then reset this state so it can try to build a new channel.
-	if c.isTimedOut() {
+	// If this channel timed out or if the confirmed tranactions straddle the Holocene activation time
+	// return true so that the the caller can reset the state abd build a new channel.
+	if c.isTimedOut() || c.straddlesActivation(holoceneTime) {
```

Comment:
Would we care whether the condition was triggered due to timeout or straddle? If so, maybe worth logging the correct reason? (This comment is actually for line 92)

Reply:
Good call, fixed in [282740f](https://github.com/ethereum-optimism/optimism/pull/12743/commits/282740fb057ceaaed2855dcaf1e8b891afab22ed)